5.x -review

markstory opened the pull request.

josbeir committed:
Fix deprecations

josbeir committed:
Cleanup and remove deprecated methods

- Remove IndexRegistery and Introduce IndexLocatorAwareTrait
- Remove deprecated methods
- Fix CS config
- Bump stan level from 6 to 8
- Add Rector + Apply rectoring

josbeir committed:
Initial elastica:9.* upgrade

josbeir committed:
Bump CI config

Implement a fallback for ConnectionManager url usage

josbeir committed:
Fix CI

Bumb stan/phive version restraint

josbeir committed:
Bump phpunit to 11.5.X || 12.1.x
Fix 8.1 failure

josbeir committed:
Add composer dependency allowing 8.1 tests to pass
Fix connection DSN

josbeir committed:
Update README to reflect version 5.x

josbeir committed:
Create and update unit tests with goal to improve code coverage.
- We are now from around 75% to +- 88%

josbeir committed:
Rectoring updates + Test failure fixes

josbeir committed:
PHPunit 10/11 compatiblity fix

josbeir committed:
Update README to reflect changes
Update codecov ci config + update badge

josbeir committed:
Fix codecov branch url

josbeir committed:
Fix coverage generation

josbeir committed:
Fix version table

josbeir committed:
Code quality improvements
Remove pslam
Add rector
Update CI
Remove phpstan baseline and add fix some previously ignored errors

josbeir committed:
Add missing or improve tests for
- IsUnique
- DeleteQueryStrategry (also make this more robus)
- IndexLocator

josbeir committed:
Move test index file to testApp as it was in the wrong place

josbeir committed:
Enable use of custom logging engine

josbeir committed:
Fix ElasticLogger so it works with Elastica/esphp 9
Improve type hinting

josbeir committed:
Improve logging

josbeir committed:
Remove empty info from logs

josbeir committed:
Respect the contract: use QueryInterface in Index

josbeir committed:
Remove orphaned docblock line

josbeir committed:
Rename Plugin.php => ElasticSearchPlugin.php
Revert contract naming

josbeir committed:
Add deprecation for using option arrays with finder methods
Fixes #336

josbeir committed:
Add support for InvalidPropertyInterface on Document entity

josbeir committed:
Fix tests

josbeir committed:
Improve coverage of newly added InvalidPropertyInterface functionality

josbeir committed:
feat(query) add Query:trackTotalHits()
Adds support for elasticsearch track_total_hits in queries

josbeir committed:
Improve strict typing

josbeir committed:
Update readme

markstory commented on src/Datasource/Connection.php:
```diff
@@ -67,19 +74,35 @@ class Connection implements ConnectionInterface
+        }
+
+        // Allow configuration from DSN format.
+        if (isset($config['host'])) {
+            $config['hosts'] = [$config['host'] . ':' . $config['port']];
+            unset(
+                $config['host'],
+                $config['port'],
+            );
```
Thank you for keeping a backwards compatible path.

markstory commented on tests/TestCase/Datasource/Log/ElasticLoggerTest.php:
```diff
@@ -0,0 +1,484 @@
+                           isset($data['query']['match_all']);
+                }),
+                $this->anything(),
+            );
```
Sometimes it is simpler to use `ArrayLog` to capture all the log messages and then you can do assertions on the order and content.

markstory commented on tests/TestCase/Datasource/ConnectionTest.php:
```diff
@@ -53,88 +71,412 @@ public function testGetIndex()
+        $this->assertInstanceOf(CacheInterface::class, $defaultCacher);
+
+        // Test setting custom cacher
+        $mockCache = $this->createMock(CacheInterface::class);
```
You could use `Cache\Engine\ArrayEngine` in situations like this in the future.

markstory commented on tests/TestCase/TestSuite/TestFixtureTest.php:
```diff
@@ -0,0 +1,799 @@
+
+        $this->assertInstanceOf(Index::class, $result);
+        // The method should use Inflector::camelize on the table name
+        // test_articles -> TestArticles
```
Looks like some assertions missing.

markstory reviewed (commented):
Sorry, I wasn't able to get through the whole thing. If you want more detailed feedback in the future smaller changes are easier for me to review. Of what I did read, it looked good.

josbeir committed:
Refactor ElasticLoggerTest to use ArrayLog

josbeir committed:
Improve type hinting on Query::all()

josbeir committed:
Finish missing assertations in TestFixtureTest

josbeir committed:
Refactor ConnectionTest to use ArrayEngine

josbeir commented:
Thanks @markstory !
I have updated and resolved your feedback. Should we plan a release  ?

josbeir committed:
refactor: Align Marshaller with CakePHP ORM patterns

- Delegate entity/nested creation to Index classes
- Add Model.afterMarshal event and fieldValue() helper
- Add Embedded::getIndex/setIndex for marshaller delegation
- Improve dirty flag handling for clean embedded entities

josbeir commented:
Implemented functionality from #333 in https://github.com/cakephp/elastic-search/pull/338/commits/ba0fdd2c4f4ed07f8dfb1c98e26b7264c298c66e

josbeir committed:
feat: implement MappingGenerator indexes filtering

Implement the previously unused $indexes parameter in
MappingGenerator::reload() to allow selective index reloading.

* Add index name filtering logic - skip indexes not in the filter list
* Add file_exists() check before include() to prevent warnings
* Add @mixin annotation to Connection class for better IDE support

When $indexes is null, all indexes are reloaded (default behavior).
When an array is provided, only matching indexes are processed.

josbeir committed:
feat: Add FQDN class support to IndexLocator and IndexLocatorAwareTrait

This reflects newly added FQDN support in TableLocator (CakePHP 5.2.9)

josbeir committed:
ci: Add php 8.5
